opendevin/core/main.py

```diff
@@ -163,7 +163,7 @@ async def on_event(event: Event):
             raise ValueError(f'Invalid toml file, cannot read {args.llm_config}')
         llm = LLM(llm_config=llm_config)
     else:
-        llm = LLM(model=args.model_name)
+        llm = LLM(llm_config=config.get_llm_config_from_agent(args.agent_cls))
```

Comment:
This breaks `--model-name` command line argument.

Looks like we should just remove `--model-name` from CLI.

Reply:
Just fyi, did a cursory search: while `model-name` is barely found (just for processing), the shorthand `-m` is used in a couple of .py/.md files, also evaluations.

Reply:
I would agree here, been thinking about removing it before. It's a bit unexpected now too, it's the most specific but llm_config overrides it.

It affects IDE configurations and cli personal scripts. We may want to give a heads-up on slack.

Reply:
IMO `main` is okay to break without notice. Will give a heads-up on slack without @ everyone.

Release is trickier - we need to have a release note. @mamoodi Do you have an idea where we might document all breaking changes? A shared doc would suffice. Alternatively, we could add something like a `changelog.md` to document that.

Reply:
I am going to go ahead and create a changelog.md doc.